### ISSUE TITLE
Immediate crash of editor upon opening new post page (solved by updating nginx config)

### ISSUE THREAD

**Issue by Kilian** [WordPress/gutenberg#12655]
**Describe the bug**
Upon opening the "new post" page, the editor is briefly visible before being replaced by a "The editor has encountered an unexpected error." popup. There is a button "attempt recovery", which does not work.

**To Reproduce**
1. Go to  the "new post" page
2. See error

**Expected behavior**
not see error.

**Screenshots**
![selection_002](https://user-images.githubusercontent.com/41970/49609241-33da6d00-f99b-11e8-8a33-e02c2f4560eb.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 16.04
 - Browser Firefox
 - Version 63

**Additional context**
The Gutenberg editor in wordpress 5.0. I cleared my localStorage settings but the error remains.

** Copies error **
```
this.wp.editor</Hs</<@https://kilianvalkhof.com/wp-includes/js/dist/editor.min.js?ver=9.0.4:55:239952
i@https://kilianvalkhof.com/wp-includes/js/dist/vendor/lodash.min.js?ver=4.17.11:6:91
x/An.filter@https://kilianvalkhof.com/wp-includes/js/dist/vendor/lodash.min.js?ver=4.17.11:99:327
this.wp.editor</Hs<@https://kilianvalkhof.com/wp-includes/js/dist/editor.min.js?ver=9.0.4:55:239921
ph@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.6.3:97:88
eg@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.6.3:125:307
fg@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.6.3:126:168
wc@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.6.3:138:237
fa@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.6.3:137:115
gg@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.6.3:135:196
Ca@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.6.3:133:365
enqueueSetState@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.6.3:191:324
q.prototype.setState@https://kilianvalkhof.com/wp-includes/js/dist/vendor/react.min.js?ver=16.6.3:20:291
value@https://kilianvalkhof.com/wp-includes/js/dist/data.min.js?ver=4.0.1:1:13979
n/<@https://kilianvalkhof.com/wp-includes/js/dist/data.min.js?ver=4.0.1:1:10354
n@https://kilianvalkhof.com/wp-includes/js/dist/data.min.js?ver=4.0.1:1:10325
g/<.subscribe</<@https://kilianvalkhof.com/wp-includes/js/dist/data.min.js?ver=4.0.1:1:8324
b@https://kilianvalkhof.com/wp-includes/js/dist/data.min.js?ver=4.0.1:1:19536
y/</<@https://kilianvalkhof.com/wp-includes/js/dist/data.min.js?ver=4.0.1:1:5320
v/</</<@https://kilianvalkhof.com/wp-includes/js/dist/data.min.js?ver=4.0.1:1:5748
l/</</<@https://kilianvalkhof.com/wp-includes/js/dist/redux-routine.min.js?ver=3.0.3:1:9095
dispatch@https://kilianvalkhof.com/wp-includes/js/dist/data.min.js?ver=4.0.1:1:22001
a/<@https://kilianvalkhof.com/wp-includes/js/dist/redux-routine.min.js?ver=3.0.3:1:8676
r/<@https://kilianvalkhof.com/wp-includes/js/dist/redux-routine.min.js?ver=3.0.3:1:4477
r@https://kilianvalkhof.com/wp-includes/js/dist/redux-routine.min.js?ver=3.0.3:1:4451
u/<@https://kilianvalkhof.com/wp-includes/js/dist/redux-routine.min.js?ver=3.0.3:1:4406
j/</<@https://kilianvalkhof.com/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:2:29352
j/<@https://kilianvalkhof.com/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:2:29220
f@https://kilianvalkhof.com/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:1:20743
```

**Comment by danielbachhuber** [comment]
Hi @Kilian,

Thanks for the report! I'm unable to reproduce this in WordPress 5.0.

A couple follow-up questions:
* Are you running any other plugins? Can you try deactivating them to see if one might be the cause?
* Can you open Developer Tools in your browser to see if any script requests (or other HTTP requests) may have failed?

**Comment by AlirezaSedghi** [comment]
Hi, I also have this problem.
Firefox Developer Console display this:
```
Unhandled promise rejection Error: "[object Object]"          wp-polyfill.min.js:2:29718
	r wp-includes/js/dist/redux-routine.min.js?ver=3.0.3:1:8606
	r wp-includes/js/dist/redux-routine.min.js?ver=3.0.3:1:8564
	j wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:2:29352
	j wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:2:29220
	f wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:1:20743

T/</n<           wp-includes/js/dist/vendor/wp-polyfill.min.js:2:29718
[89]</n.exports  wp-includes/js/dist/vendor/wp-polyfill.min.js:1:25719
T/<              wp-includes/js/dist/vendor/wp-polyfill.min.js:2:29595
[51]</n.exports  wp-includes/js/dist/vendor/wp-polyfill.min.js:1:16271
<anonymous>      wp-includes/js/dist/vendor/wp-polyfill.min.js:1:30590
m                wp-includes/js/dist/vendor/wp-polyfill.min.js:1:30435
x                wp-includes/js/dist/vendor/wp-polyfill.min.js:1:30455
```

**Comment by danielbachhuber** [comment]
Thanks for the additional data, @Kilian.

> Edit: On chrome 71, the editor crashes still but the actual error is different:
>
>      TypeError: Cannot read property 'show_ui' of undefined

This is helpful. It seems like there are some additional recent reports in https://github.com/WordPress/gutenberg/issues/9556#issuecomment-443546483

Also, #8802 is an earlier report.

The immediate problem is that Gutenberg is trying to access data about taxonomies, but the `show_ui` field isn't available. There's likely some deeper root cause though.

Can you take a look at your "Network" tab in your browser's Developer Tools to see if any API requests are failed? And, if so, share screenshots of those?

> Furthermore i get a 403 on `https://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data`

This is a Jetpack issue and likely unrelated (but you never know).

**Comment by Kilian** [comment]
Other than the 403 mentioned there are no other failed requests.

**Comment by designsimply** [comment]
I noticed that https://github.com/WordPress/gutenberg/issues/8899#issuecomment-413318042 mentions the `Cannot read property 'show_ui' of undefined` error was solved by changing the permalink structure (and eventually the rewrite rules on the server). @Kilian can you note whether you're using a custom permalink and what that is? If you update the permalink settings, does it make the problem go away? See https://github.com/WordPress/gutenberg/issues/8899#issuecomment-413318042

**Comment by mdparker** [comment]
I'm seeing the same errors as @Kilian in Chrome. Disabling my plugins did not solve the problem for me.

**Comment by igortitarenko** [comment]
I got rid of
"The editor has encountered an unexpected error."
error by commenting out
"try_files $uri $uri/ /website/index.php?q=$uri$args;"
in my Nginx configuration.

But Gutenberg still doesn't work.

`Unhandled promise rejection Error: "[object Response]"

    r2Redux

    	j https://example.com/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:2
    	j https://example.com/barbosha/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:2
    	f https://example.com/barbosha/wp-includes/js/dist/vendor/wp-polyfill.min.js?ver=7.0.0:1

wp-polyfill.min.js:2:29718`

**Comment by igortitarenko** [comment]
And this is what I see in the logs after I turn ModSecurity on and click "Attempt Recovery" button

```
2018/12/08 19:45:29 [info] 12659#0: *1 ModSecurity: Warning. Matched "Operator `Pm' with parameter `document.cookie document.write .parentnode .innerhtml window.location -moz-binding <!-- --> <![cdata[' against variable `ARGS:json.content' (Value: `\xd0\x94\xd0\xbe\xd0\xb1\xd1\x80\xd0\xbe \xd0\xbf\xd0\xbe\xd0\xb6\xd0\xb0\xd0\xbb\xd0\xbe\xd0\xb2\xd (530 characters omitted)' ) [file "/etc/nginx/owasp-modsecurity-crs/rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf"] [line "262"] [id "941180"] [rev "2"] [msg "Node-Validator Blacklist Keywords"] [data "Matched Data: <!-- found within ARGS:json.content: \xd0\x94\xd0\xbe\xd0\xb1\xd1\x80\xd0\xbe \xd0\xbf\xd0\xbe\xd0\xb6\xd0\xb0\xd0\xbb\xd0\xbe\xd0\xb2\xd0\xb0\xd1\x82\xd1\x8c \xd0\xb2 WordPress. \xd0\xad\xd1\x82\xd0\xbe \xd0\xb2\xd0\xb0\xd1\x88\xd0\xb0 \xd0\xbf\xd0\xb5\xd1\x80\xd0\xb2\xd0\xb0\xd1\x8f \xd0\xb7\xd0\xb0\xd0\xbf\xd0\xb8\xd1\x81\xd1\x8c. \xd0\x9e\xd1\x82\xd1\x80\xd0\xb5\xd0\xb4\xd0\xb0\xd0\xba\xd1\x82\xd0\xb8\xd1\x80\xd1\x83\xd0\xb9\xd1\x82\xd0\xb5 \xd0\xb8\xd0\xbb\xd0\xb8 \xd1\x83\xd0\xb4\xd0\xb0\xd0\xbb\xd0\xb8\xd1\x82\xd0\xb5 \xd0\xb5\xd1\x91, \xd0\xb7\xd0\xb0\xd1\x82\xd0\xb5\xd0\xbc \xd0\xbf\xd0\xb8\xd1\x88\xd0\xb8\xd1\x82\xd0\xb5!<!--pagesep-->\x0a\x0a&nbsp;"] [severity "2"] [ver "OWASP_CRS/3.0.0"] [maturity "1"] [accuracy "8"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-xss"] [tag "OWASP_CRS/WEB_ATTACK/XSS"] [tag "WASCTC/WASC-8"] [tag "WASCTC/WASC-22"] [tag "OWASP_TOP_10/A3"] [tag "OWASP_AppSensor/IE1"] [tag "CAPEC-242"] [hostname "IP"] [uri "/subdir/wp-json/wp/v2/posts/1"] [unique_id "154429832967.728779"] [ref "o98,4v13,192t:utf8toUnicode,t:urlDecodeUni,t:htmlEntityDecode,t:jsDecode,t:cssDecode,t:lowercase,t:removeNulls"], client: IP, server: example.com, request: "POST /subdir/wp-json/wp/v2/posts/1?_locale=user HTTP/2.0", host: "example.com", referrer: "https://example.com/subdir/wp-admin/post.php?post=1&action=edit"
2018/12/08 19:45:29 [info] 12659#0: *1 ModSecurity: Access denied with code %d (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/etc/nginx/owasp-modsecurity-crs/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "36"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver ""] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "IP"] [uri "/subdir/wp-json/wp/v2/posts/1"] [unique_id "154429832967.728779"] [ref ""], client: IP, server: example.com, request: "POST /subdir/wp-json/wp/v2/posts/1?_locale=user HTTP/2.0", host: "example.com", referrer: "https://example.com/subdir/wp-admin/post.php?post=1&action=edit"
2018/12/08 19:45:29 [warn] 12659#0: *1 [client IP] ModSecurity: Warning. Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/etc/nginx/owasp-modsecurity-crs/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "36"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver ""] [maturity "0"] [accuracy "0"] [hostname "IP"] [uri "/subdir/wp-json/wp/v2/posts/1"] [unique_id "154429832967.728779"] [ref ""], client: IP, server: example.com, request: "POST /subdir/wp-json/wp/v2/posts/1?_locale=user HTTP/2.0", host: "example.com", referrer: "https://example.com/subdir/wp-admin/post.php?post=1&action=edit"
2018/12/08 19:45:29 [info] 12659#0: *1 ModSecurity: Warning. Matched "Operator `Ge' with parameter `5' against variable `TX:INBOUND_ANOMALY_SCORE' (Value: `5' ) [file "/etc/nginx/owasp-modsecurity-crs/rules/RESPONSE-980-CORRELATION.conf"] [line "61"] [id "980130"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Inbound Score: 5 - SQLI=0,XSS=5,RFI=0,LFI=0,RCE=0,PHPI=0,HTTP=0,SESS=0): Node-Validator Blacklist Keywords'"] [data ""] [severity "0"] [ver ""] [maturity "0"] [accuracy "0"] [tag "event-correlation"] [hostname "IP"] [uri "/subdir/wp-json/wp/v2/posts/1"] [unique_id "154429832967.728779"] [ref ""] while logging request, client: IP, server: example.com, request: "POST /subdir/wp-json/wp/v2/posts/1?_locale=user HTTP/2.0", host: "example.com", referrer: "https://example.com/subdir/wp-admin/post.php?post=1&action=edit"
```

**Comment by umakantsharmacedcoss** [comment]
same error, if you are trying to **create a new block** from **manage all reusable blocks**

wp-admin/post-new.php?post_type=wp_block

**Comment by Kilian** [comment]
**Fixed**

tldr: if you run Nginx, you might need to update your nginx config.


> I have the same problem. With posts only. Pages work fine.

This made me check. The editor works fine for pages, it's just the posts that crash.

>  [...] error was solved by changing the permalink structure (and eventually the rewrite rules on the server)

I indeed have a custom permalink structure: `/%year%/%category%/%postname%/`, with what was then the default digitalocean configuration/tutorial for wordpress with permalinks on Nginx. I checked [a current tutorial](https://www.digitalocean.com/community/tutorials/how-to-install-wordpress-with-lemp-on-ubuntu-18-04) and I noticed that I had `$is_args$args` missing from the main location block:

```
    location / {
        try_files $uri $uri/ /index.php$is_args$args;
    }
```

After adding this to the end of /index.php, the editor now works for posts too!

**Comment by danielbachhuber** [comment]
Thanks for the update, @Kilian.

The specific issue you found make sense to me. If your Nginx configuration was incorrect, then `?context=edit` wasn't passing through to WordPress which meant you'd get a Taxonomy object with `?context=view` (and, thus, missing its `visibility` attribute).

It was only a problem with Posts for you because you don't have any taxonomies assigned to Pages.

**Comment by pojke** [comment]
Unfortunately, my issue is different so I'll log a new bug

**Comment by stz184** [comment]
I can confirm the issue, same browser version and OS.
The nginx configuration is OK, already checked this.

**Comment by micele** [comment]
Exact same issue here. Still remains with WP 5.0.2.

FreeBSD 11.1
Apache 2.4.37
Chrome and FF

**Comment by designsimply** [comment]
@AlirezaSedghi the error in your case is a little different. Can you please report your case at https://github.com/WordPress/gutenberg/issues/12982 and include your WordPress, OS, and browser versions?

@mdparker @igortitarenko @digitalformula if you haven't already, please try the solution posted at https://github.com/WordPress/gutenberg/issues/12655#issuecomment-445816958 if you are using nginx.

@umakantsharmacedcoss  issues with `wp-admin/post-new.php?post_type=wp_block` may be different. Could you please post the details of your case at https://github.com/WordPress/gutenberg/issues/12324 ?

@stz184 @micele I would like to close this issue since it's getting very long and because one solution that is working for some was posted so it's possible your issue may be different. Would you mind please opening a new issue with the details of your case including: the error you see on screen, the error you see in your browser console, WordPress version, OS & browser version, note whether you've tested with all plugins deactivated, and your the location rule from your nginx config (or the corresponding rule in Apache if you are using Apache)?

**Comment by mdparker** [comment]
@designsimply No nginx here. It's Apache 2.4.17 on FreeBSD.

**Comment by digitalformula** [comment]
@designsimply Apologies for the delayed response - I missed the notification that you tagged people.

Yes, I'm running Nginx.  I've added `$is_args$args` in the relevant place, as outlined by the linked comment and restarted Nginx.  Issue is still there.

My permalink structure is the same as the one in the linked comment i.e. `/%year%/%category%/%postname%/` (it was `/%postname%/` but I changed it for testing).

No difference for me, unfortunately.  Exact same error manifested in the exact same way.  The message saying welcome to the world of blocks appears momentarily, then is replaced by `The editor has encountered an unexpected error.`

**Comment by samdufeu** [comment]
> 
> 
> **Fixed**
> 
> tldr: if you run Nginx, you might need to update your nginx config.
> 
> > I have the same problem. With posts only. Pages work fine.
> 
> This made me check. The editor works fine for pages, it's just the posts that crash.
> 
> > [...] error was solved by changing the permalink structure (and eventually the rewrite rules on the server)
> 
> I indeed have a custom permalink structure: `/%year%/%category%/%postname%/`, with what was then the default digitalocean configuration/tutorial for wordpress with permalinks on Nginx. I checked [a current tutorial](https://www.digitalocean.com/community/tutorials/how-to-install-wordpress-with-lemp-on-ubuntu-18-04) and I noticed that I had `$is_args$args` missing from the main location block:
> 
> ```
>     location / {
>         try_files $uri $uri/ /index.php$is_args$args;
>     }
> ```
> 
> After adding this to the end of /index.php, the editor now works for posts too!

Thanks! 


It's fairly simple to assume its a URL re-write issue when changing the default behavior in WordPress can resolve this issue. For those on Apache, you will need a similar mod. 

Perhaps the codex needs updating to include this? [Link](https://codex.wordpress.org/Nginx)
